### ISSUE TITLE
Tabs stop grouping and moving to the beginning

### ISSUE THREAD

**Issue by leo8nardo8** [jdhayford/acid-tabs-extension#20]
Hey,
It's feeling a little buggy last days.

Tabs are still getting tagged but it no longer grouping by category and grouping at the begins.

![behavior](https://user-images.githubusercontent.com/32337025/132026388-6bd8990c-3082-423e-a30a-60e0aba59349.png)
![wanted](https://user-images.githubusercontent.com/32337025/132026393-22f115e1-90a7-4460-b571-93cee5c23d85.png)

**Comment by dhallam** [comment]
I'm seeing issues too. In Chrome 92.0.4515.159 on Ubuntu 21.04 it's no longer bunching up tabs into their groups. It's adding tabs to a new instance of a group so all the groups have just 1 tab in them. 

![Selection_697](https://user-images.githubusercontent.com/117853/132811371-2246ef32-5d19-430a-b15b-dc1a8838ef00.png)

**Comment by dhallam** [comment]
NB: I just resolved my above issue by copying the configuration out of bulk edit into a text editor, disabling the extension, manually ungrouping everything, re-enabling the extension, and pasting the configuration into bulk edit. Everything came back as expected.

**Comment by leo8nardo8** [comment]
It works! Thanks.

**Comment by jdhayford** [comment]
@dhallam @leo8nardo8 Glad the bulk edit ability helped you somewhat here. I have not experienced this problem, did either of you see anything that could help identify the cause?

**Comment by 0xC0ncord** [comment]
> @dhallam @leo8nardo8 Glad the bulk edit ability helped you somewhat here. I have not experienced this problem, did either of you see anything that could help identify the cause?

I also see this behavior. I see it most often when re-opening an older or existing session in Chromium, which doesn't cause all the tabs to load immediately. Unloaded tabs seem to be unsorted, and when they are loaded they will move into a new group. Loading all of the tabs causes them to move into this new group of loaded tabs until every unloaded tab is loaded, and the group containing the unloaded tabs is emptied.

**Comment by bsell93** [comment]
@jdhayford I think I figured out the issue here! It turns out Chrome `sync` storage has a max number of items. [Google Docs](https://developer.chrome.com/docs/extensions/reference/storage/#property-sync)
<img width="740" alt="image" src="https://user-images.githubusercontent.com/5335177/187222547-17becb56-f3a3-4ffe-b612-a9c800f90d9f.png">

I just started having this problem this morning. So after a bunch of console logs, 
I ended up logging `chrome.runtime.lastError` in the `get` method and got this in my console numerous times
`Unchecked runtime.lastError: MAX_ITEMS quota exceeded`

Investigating a potential fix. My guess is we need to clear the sync storage after a certain amount of time or on certain events.

Let me know if you have any thoughts @jdhayford

**Comment by jdhayford** [comment]
> @jdhayford this SO post suggests storing in local. Which we could for just the `Group` data. I don't see much gain in storing those key/value pairs in sync storage. Benefit of that is we can add "unlimited" storage request for local. We really should be cleaning up the keys periodically so I feel like the local storage option alone is the best solution. Would be a good solution until we find the best way to clean up the unused keys https://stackoverflow.com/a/29732624/4965713

Yeah great finds Bryant! I too recently discovered that some items are definitely "leaking" so to speak. I was originally considering the periodic cleanup job (comparing against window ids that still exist to avoid removing relevant items). Switching the groups to local makes perfect sense though and should also help give us much bigger margin.

I think creating a simple cleanup task will be good as a first step, that way if we do switch to local we can know that we aren't leaving user's sync storage too full in case we want to use that for later functionality.
